### DISCUSSION TITLE
How to use the common reference string in groth16 to generate random numbers?

### DISCUSSION THREAD

**Discussion by 2pir2** [Consensys/gnark#1310] [Q&A]
I am trying to generate a sequence of random number as inputs. I am wondering if I can just use the common reference string in groth16 as the source of random seed.

**Comment by ivokub** [comment]
There is the `api.Commit` method which uses the SRS indirectly. It returns a single element (as internally we compute the Pedersen vector commitment which is a big MSM), but you can expand it to a sequence of random values using consecutive multiplications for example.

**Reply by ivokub** [reply]
We use it in gnark quite extensively actually, it allows to perform efficient random evaluations. But the verifier work becomes less efficient the more times `api.Commit` is called. For that we have implemented a multicommit method which expands the single `api.Commit` output to a series and provides them internally to callbacks. See https://pkg.go.dev/github.com/consensys/gnark@v0.11.0/std/multicommit and the example within.
